pkg/apis/run/v1beta1/customrunstatus_types.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2020 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"encoding/json"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+)
+
+// This package exists to avoid an import cycle between v1beta1 and v1beta1.
+// It contains common definitions needed by v1beta1.CustomRun and v1beta1.PipelineRun.
+
+// +k8s:deepcopy-gen=true
+
+// CustomRunStatus defines the observed state of CustomRun
+type CustomRunStatus struct {
+	duckv1.Status `json:",inline"`
+
+	// CustomRunStatusFields inlines the status fields.
+	CustomRunStatusFields `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+
+// CustomRunStatusFields holds the fields of CustomRun's status.  This is defined
+// separately and inlined so that other types can readily consume these fields
+// via duck typing.
+type CustomRunStatusFields struct {
+	// StartTime is the time the build is actually started.
+	// +optional
+	StartTime *metav1.Time `json:"startTime,omitempty"`
+
+	// CompletionTime is the time the build completed.
+	// +optional
+	CompletionTime *metav1.Time `json:"completionTime,omitempty"`
+
+	// Results reports any output result values to be consumed by later
+	// tasks in a pipeline.
+	// +optional
+	Results []CustomRunResult `json:"results,omitempty"`
+
+	// RetriesStatus contains the history of CustomRunStatus, in case of a retry.
+	// +optional
+	RetriesStatus []CustomRunStatus `json:"retriesStatus,omitempty"`
+
+	// ExtraFields holds arbitrary fields provided by the custom task
+	// controller.
+	ExtraFields runtime.RawExtension `json:"extraFields,omitempty"`
+}
+
+// CustomRunResult used to describe the results of a task
+type CustomRunResult struct {
+	// Name the given name
+	Name string `json:"name"`
+	// Value the given value of the result
+	Value string `json:"value"`
+}
+
+var customRunCondSet = apis.NewBatchConditionSet()
+
+// GetCondition returns the Condition matching the given type.
+func (r *CustomRunStatus) GetCondition(t apis.ConditionType) *apis.Condition {
+	return customRunCondSet.Manage(r).GetCondition(t)
+}
+
+// InitializeConditions will set all conditions in customRunCondSet to unknown for the PipelineRun
```

Comment:
IIUC, CustomRun is able to run by itself right? So this func is not only for PipelineRun?

Reply:
yes, function sets everything to unknown for the PipelinRun

Reply:
Yeah it can be used by PipelineRun but it is not only for PipelineRun (The CustomRun can run by itself, the controller for the CustomRun can use this func to initialize the conditions), which makes the comment not accurate. Deleting "for the PipelineRun" is more proper in this case. @jerop to confirm.

Reply:
Good catch @XinruZhang - it should have been `Run` in v1alpha1 and should be `CustomRun` here

(must be a copy paste mistake in v1alpha1)

Reply:
fixed, thank you @XinruZhang @jerop